src/Gruntfile.js

```diff
@@ -309,6 +310,55 @@ module.exports = function (grunt) {
     });
   });
 
+  function hasBOM(buf) {
+    return buf.length > 2 && buf[0] == 0xef && buf[1] === 0xbb && buf[2] === 0xbf;
+  }
+
+  function parseJSON(text) {
+    try {
+      return {
+        json: JSON.parse(text),
+      };
+    } catch(err) {
+      return {
+        error: err.message,
+      };
+    }
+  }
+
+  // Async task structure: https://gruntjs.com/creating-tasks
+  grunt.registerTask("validate_links", "Check if links return 200 and valid json", async function () {
```

Comment:
If someone adds a new schema in their PR, the URL to that schema doesn't exist on schemastore.org yet. Would this task then fail?

Reply:
Yes, but the task is not integrated into any CI...
So its a limited solution that just provides some information after publishing :)
Can change this around if you want...

Sometimes schema's have a real identifying url (the $id)... However sometimes the url in the catalog just represents a temporary source (raw github). The raw github urls pojects might not even be aware that the file location is important within their project.

Maybe its an idea to cache all schema's into the SchemaStore besides having an $id and source url. This would allow validating the schema's locally and offers a solution when schema accidentally become unavailable as raw github sources. Maybe its also important to assign a github project as the source of maintenance for a schema? Now its difficult to find out who is the contact for schema feedback...